tests/test_solvers/test_openclaw_config.py

```
@@ -0,0 +1,221 @@
+        # A placeholder apiKey trips OpenClaw's auth resolver and breaks the
+        # synthetic local-key path -- the field must be absent, not blank.
+        config = _build_openclaw_config("http://localhost:9000", "nvidia/m", api_key="")
+        assert "apiKey" not in config["models"]["providers"]["nvidia"]
+
+    def test_api_key_included_when_provided(self) -> None:
+        config = _build_openclaw_config("http://localhost:9000", "nvidia/m", api_key="sk-real")
+        assert config["models"]["providers"]["nvidia"]["apiKey"] == "sk-real"
+
+    def test_optional_token_limits_included_only_when_set(self) -> None:
+        without = _build_openclaw_config("u", "nvidia/m")["models"]["providers"]["nvidia"]["models"][0]
+        assert "contextWindow" not in without and "maxTokens" not in without
```

Ruff (0.15.15)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)